evm/src/cpu/kernel/interpreter.rs

```diff
@@ -54,7 +36,8 @@ pub(crate) struct Interpreter<'a> {
     jumpdests: Vec<usize>,
     offset: usize,
     pub(crate) stack: Vec<U256>,
```

Comment:
Let's add a TODO to remove stack + code? Since they're also segments within `memory`. Using the segments in `memory` will mean we can use `mload_general` to load constants from code, or for stack-based arrays etc.

Reply:
Good point! Do we need to maintain a pointer to the top of the stack? 
Also, for the code, would the offset become a context+offset? Like right now the kernel code is at the start of the `Code` segment in the first context. So if we want to jump to a kernel function from another context, we would have to jump to the first context? Or would we write the kernel code at every context?

Reply:
Yeah I think we'll need to add a stack pointer register in the CPU, though our assembly shouldn't need to know about it (unless we wanted stack-based arrays, but doesn't really seem necessary).

I think the kernel code can live in context 0 only. When we jump to kernel code, I think it may be easiest/cheapest to leave the CPU's context register alone, so it still holds the (userland) caller's context. Then when we fetch an instruction, we'd load from context `is_user_mode * context`. Does that sound okay? cc @nbgl who is starting on some related work.

Reply:
Yes sounds good!